blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -392,6 +392,34 @@ namespace blueprint
          **/
         BundleEvalCallback afterBundleEval;
 
+        //==============================================================================
+        using EventSubscriptionToken = int;
+
+        /**
+         * Subscribe to a given event from js. This should be called in your beforeBundleEval
+         * callback as the set of listeners/subscriptions will be cleared on bundle reloads.
+         * Returns an EventSubscriptionToken instance which callers may store and later use to unsubscribe()
+         * a previously registered event handler.
+         * Note, callers should take care to avoid callback loops between subscribe and dispatchEvent etc.
+         **/
+        EventSubscriptionToken subscribe(const juce::String &eventType, juce::var::NativeFunction handler)
+        {
+            JUCE_ASSERT_MESSAGE_THREAD
+            jassert(handler);
+
+            engine.invoke("__BlueprintNative__.subscribe", eventType, std::move(handler));
```

Comment:
Oh, oops, I missed this: shouldn't this be `return engine.invoke("__BlueprintNative__.subscribe", eventType, std::move(handler));` Right now this will just drop javascript's understanding of the token and give out a 0 every time

Reply:
Arrrg sorry! I altered that earlier whilst testing something. Good job you're on the ball Nick. I'm clearly a bit sleepy today. Will fix.

Reply:
OK resolved and tested it actually works again.